decorate.rb

```diff
@@ -0,0 +1,31 @@
+require_relative 'nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  MAX_NAME_LENGTH = 10
+
+  def correct_name
+    name = @nameable.correct_name
+    if name.length > MAX_NAME_LENGTH
+      name.slice(0, MAX_NAME_LENGTH)
+    else
+      name
+    end
+  end
```

Comment:
- [ ] It would be better to move the CapitalizeDecorator and TrimmerDecorator classes into separate files this will make your project follow best practices